### PR TITLE
[RUNTIME][REFACTOR] Re-organize Containers into SubFolders

### PR DESCRIPTION
**Pull request by ZihengJiang**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

File: include/tvm/runtime/container/shape_tuple.h [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+#include "./base.h"
+
+namespace tvm {
+namespace runtime {
+
+class ShapeTupleObj : public Object {
+ public:
+  using index_type = int64_t;
+  index_type* data;
+  uint64_t ndim;
```
Comment (tqchen) [inline include/tvm/runtime/container/shape_tuple.h:39]: use size and make interface as consistent with STL vector<int> as ppossible.

File: include/tvm/runtime/container/shape_tuple.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+#include <utility>
+#include <vector>
+
+#include "./base.h"
+
+namespace tvm {
+namespace runtime {
+
+class ShapeTupleObj : public Object {
```
Comment (tqchen) [inline include/tvm/runtime/container/shape_tuple.h:35]: need document of the classes

File: include/tvm/runtime/container/shape_tuple.h [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  ShapeTuple() : ShapeTuple(std::vector<index_type>()) {}
+  template <typename Iterator>
+  ShapeTuple(Iterator begin, Iterator end) : ShapeTuple(std::vector<index_type>(begin, end)) {}
+  ShapeTuple(std::initializer_list<index_type> shape) : ShapeTuple(shape.begin(), shape.end()) {}
+
+  ShapeTuple(std::vector<index_type> shape);  // NOLINT(*)
+
+  index_type operator[](size_t idx) const { return get()->data[idx]; }
+  index_type at(size_t idx) const { return get()->data[idx]; }
+  index_type* data() { return get()->data; }
```
Comment (tqchen) [inline include/tvm/runtime/container/shape_tuple.h:74]: only support constant index_type to keep shape tuple immutable

File: include/tvm/runtime/container/shape_tuple.h [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  size_t ndim() const { return get()->ndim; }
+  size_t size() const { return get()->ndim; }
```
Comment (tqchen) [inline include/tvm/runtime/container/shape_tuple.h:76]: Only use size is fine.

**Comment by tqchen** [comment]
cc @junrushao1994 @yzhliu

**Comment by junrushao1994** [comment]
Overall looks good, but not familiar with this CI failure

**Review by tqchen — APPROVED** [review]

**Pull request merged**